From c38b9387d01af3a3e4e270219fe5e9f524a8c6de Mon Sep 17 00:00:00 2001
From: tenzap <fabstz-it@yahoo.fr>
Date: Mon, 22 Nov 2021 08:20:47 +0100
Subject: [PATCH] add missing header: cstdlib

for compilation on older systems (like macos 10.11)

Needed for std::free() & std::calloc() on older systems [1]
That file is included indirectly on newer systems.

[1] https://en.cppreference.com/w/cpp/memory/c/free
---
 src/fuzzer/mem_pool.cpp                | 1 +
 src/lib/compat/sodium/sodium_utils.cpp | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/fuzzer/mem_pool.cpp b/src/fuzzer/mem_pool.cpp
index 722746288..c5532539b 100644
--- src/fuzzer/mem_pool.cpp
+++ src/fuzzer/mem_pool.cpp
@@ -12,6 +12,7 @@
 #include <utility>
 
 #include <stdlib.h>
+#include <cstdlib>
 
 namespace {
 
diff --git a/src/lib/compat/sodium/sodium_utils.cpp b/src/lib/compat/sodium/sodium_utils.cpp
index 3f0a6c84e..f3908f38d 100644
--- src/lib/compat/sodium/sodium_utils.cpp
+++ src/lib/compat/sodium/sodium_utils.cpp
@@ -11,6 +11,7 @@
 #include <botan/internal/os_utils.h>
 #include <botan/internal/ct_utils.h>
 #include <botan/loadstor.h>
+#include <cstdlib>
 
 namespace Botan {
 
-- 
2.30.2

